lm-simulator/lm_simulator/crud.py

```diff
@@ -50,6 +51,33 @@ async def list_licenses(session: Session) -> List[LicenseRow]:
     return [LicenseRow.model_validate(license) for license in db_licenses]
 
 
+async def list_licenses_by_server_type(session: Session, server_type: LicenseServerType) -> List[LicenseRow]:
+    """
+    List all Licenses in the database by server type.
+    """
+    query = await session.execute(select(License).where(License.license_server_type == server_type))
+    db_licenses = query.scalars().all()
+    return [LicenseRow.model_validate(license) for license in db_licenses]
+
+
+async def read_license_by_name(session: Session, license_name: str) -> Optional[LicenseRow]:
```

Comment:
That's a good point.
Before adding the validation, it was returning `None` if the license was not found.
But since it returns something or raises an exception  I'll remove the `Optional` then.
Thanks!